tests/foreman/api/test_permission.py

```diff
@@ -41,17 +63,17 @@ def test_search_by_name(self, permission_name):
         is the one searched for.
 
         """
+        permission_name = random.choice(self.permission_names)
         if (
                 permission_name == 'power_compute_resources_vms' and
                 bz_bug_is_open(1198731)):
-            self.skipTest('BZ 1198731 is open.')
+            permission_name = random.choice(self.permission_names)
```

Comment:
subTest will definitely help here, the reason that I decided to just select a random one is because now the test_search_permissions will show what have been added and removed.

But it will not be able to check that a permission is being returned more than once. If we assert and something fail it will stop the test and will not show other failures (if present). I will refactor both tests to grab the results and fail if found something.